cloudprovider/aws/aws_provider.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/rancher/agent/utilities/config"
+	"os"
+	"path"
+	"time"
+)
+
+const (
+	regionLabel           = "ec2.region"
+	availabilityZoneLabel = "ec2.availability_zone"
+	location              = "aws"
+	infoFile              = "info.json"
+	tempFile              = "temp.json"
+)
+
+type Info struct {
```

Comment:
I've changed my mind about this comment because I just got to the collector part and having a common struct for the provider and the collector is the right solution. But it shouldn't be here. It should be in the collector package and imported here.